### PR TITLE
[ntuple] Show all fields unless a model is specified

### PR DESCRIPTION
**Pull request by enirolf**
This PR introduces a new format option for `RNTupleReader::Show()` that shows all fields if no model is specified, and only the fields present in the model if there is one. It also changes the default show format to this option.

## Checklist:

- [x] tested changes locally
- [x] updated the docs (if necessary)

### CODE REVIEW

**Comment by enirolf** [comment]
I realize this commit also includes some changes to unrelated parts of the code made by clang-format. Let me know if I should revert those/put them in a separate commit.

**Comment by bellenot** [comment]
@phsft-bot build

**Review by jalopezg-git — COMMENTED** [review]
I think it should be enough if, instead of defining a new option for `Show()`, we adjust the current behaviour of `kCurrentModelJSON` to trigger model generation from the ntuple descriptor if there’s no current model.

Rationale: this is the current behaviour of `LoadEntry()`; why should `Show()` behave differently?  What do you think, @jblomer?

**Comment by enirolf** [comment]
> I think it should be enough if, instead of defining a new option for `Show()`, we adjust the current behaviour of `kCurrentModelJSON` to trigger model generation from the ntuple descriptor if there’s no current model.

Jakob and I had a quick chat yesterday and the conclusion was that this would indeed make the most sense. I thought initially that this meant we could get rid of the format option altogether, but now I think it might be good to keep the `kCurrentModelJSON` option in a model is present but the user would still like to output all fields.

**Review by jblomer — CHANGES_REQUESTED** [review]
Very nice!

I think in the `ntuple_show` unit test, we should suppress the clang-format changes for the expected strings (`// clang-format off`). Otherwise it will be difficult to adjust the strings in the future.

File: tree/ntuple/v7/src/RNTuple.cxx [hunk 1]
```diff
@@ -251,20 +238,20 @@ void ROOT::Experimental::RNTupleReader::Show(NTupleSize_t index, const ENTupleSh
...
       reader = GetDisplayReader();
       entry = reader->GetModel()->GetDefaultEntry();
       // Fall through
    case ENTupleShowFormat::kCurrentModelJSON:
       if (!entry) {
-         output << "{}" << std::endl;
-         break;
+         reader = GetDisplayReader();
+         entry = reader->GetModel()->GetDefaultEntry();
       }
```
Comment (jalopezg-git) [inline tree/ntuple/v7/src/RNTuple.cxx:250]: Alternatively, how about removing the `if` statement in lines 237-239 and directly initializing `entry` in line 236 as in
```c++
REntry *entry = GetModel()->GetDefaultEntry();
```

I think this is probably preferable, as `RNTupleReader::LoadEntry()` will initialize `fModel` from the ntuple descriptor anyways.
In the current state, a call to `Show()` followed by `LoadEntry()` triggers the generation of the model twice (once for `fDisplayReader.fModel`, and once for `fModel` itself).

Anything against proceeding this way?

Comment (enirolf) [reply tree/ntuple/v7/src/RNTuple.cxx:250]: Ah, that makes sense. Then I suppose lines 247-250 can also be removed right?

**Review by jalopezg-git — APPROVED** [review]
LGTM! @jblomer?  Consider also squashing on merge :-).

**Comment by jalopezg-git** [comment]
@phsft-bot build

**Review by jblomer — APPROVED** [review]

**Pull request merged**